smol/utils/cluster/numthreads.py

```diff
@@ -1,9 +1,15 @@
 """Simple mixin class that has an attributed/method that runs in parallel."""
 
 
+import os
+import warnings
+
 from .._openmp_helpers import _openmp_effective_numthreads
 
-DEFAULT_NUM_THREADS = _openmp_effective_numthreads(n_threads=2)
+if os.getenv("OMP_NUM_THREADS") is not None:
+    DEFAULT_NUM_THREADS = _openmp_effective_numthreads()
+else:
```

Comment:
Also, could you please add a little bit of explanation about how the case is handled when OMP_NUM_THREADS is not set, but the system has only one core available?

Reply:
That case is handled in `_openmp_effective_numthreads()`. If openmp is not available it should return 1

Reply:
Thanks! Resolved, feel free to merge!